File: 0144-binary-tree-preorder-traversal/solution.cpp

class Solution {
public:
    void display(TreeNode* root, vector<int>& v) // left ORDER DISPLAY
    {
        if (root == nullptr)
            return;
        v.push_back(root->val);
        display(root->left,v);
        display(root->right,v);
    }
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> v;
        display(root, v);
        return v;
    }
};

File: 0144-binary-tree-preorder-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
       public List<Integer> preorderTraversal(TreeNode root){
        List<Integer> res= new ArrayList<>();
        if(root==null){
            return res;
        }
        Stack<TreeNode>st= new Stack<>();
        st.push(root);
        while (!st.empty()) {
            TreeNode ts= st.pop();
            res.add(ts.val);
            if(ts.right!=null)st.push(ts.right);
            if(ts.left!=null)st.push(ts.left);
            
        }
        return res;
    }
}
